### DISCUSSION TITLE
Kotlinx Serialization for better navigation

### DISCUSSION THREAD

**Discussion by leonardovallem** [JuulLabs/kable#788] [Ideas]
Starting from 2.8.0, the Compose Navigation library now allows serializable classes to be used as destinations. So, to allow, for example, an instance of PlatformAdvertisement to be used as a navigation argument, it would have to be marked with @Serializable.

**Comment by twyatt** [comment]
I like the idea. I'll need to find some time to investigate if it is possible (I'm not certain if Android's advertisement objects, which Kable wraps, can be serialized).

**Reply by leonardovallem** [reply]
I was trying to implement it locally to check if it was possible. Turns out, I ran into some API issues. Not sure if it was a skill issue though.

However, I found out about this library https://github.com/chRyNaN/serialization-parcelable, that seems to be able to serialize Parcelable objects. So in a way or another, it should be possible.

**Comment by twyatt** [comment]
Will probably be a while before I can get to it, but tracking this possible work effort in #861.
